### PR TITLE
Publish Versioned Docs (3.1)

### PR DESCRIPTION
**Pull request by jharwig**
- [x] joeferner
- [x] diegogrz
- [ ] mwizeman sfeng88
- [x] joeybrk372 rygim jharwig EvanOxfeld

Change travis to add a job to build docs, and publish to s3 bucket docs-visallo-com. Only the branches defined in travis.yml will build.

Still have to adjust DNS from ghpages to s3.

Available at [http://docs-visallo-com.s3-website-us-east-1.amazonaws.com](http://docs-visallo-com.s3-website-us-east-1.amazonaws.com) for checking. There's a version selector in the top right corner.

Testing Instructions: None

Points of Regression: None

NO CHANGELOG

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,13 +2,17 @@ sudo: false
...
+    - $HOME/.npm
+    - $HOME/.yarn-cache
 env:
   - CC=clang CXX=clang++
-script: mvn -B -fae test -Ptest-exclude-native -DlogQuiet
-after_success:
-  - test $TRAVIS_REPO_SLUG == "v5analytics/visallo" && test $TRAVIS_BRANCH == "master" || test 'echo $TRAVIS_BRANCH | grep -Eq /^release-.*$/' && test $TRAVIS_PULL_REQUEST == "false" && echo "<settings><servers><server><id>sonatype-nexus-snapshots</id><username>\${env.DEPLOY_USERNAME}</username><password>\${env.DEPLOY_PASSWORD}</password></server></servers></settings>" > ~/settings.xml
-  - test -f ~/settings.xml && mvn -B --settings ~/settings.xml -f root/pom.xml deploy
-  - test -f ~/settings.xml && mvn -B --settings ~/settings.xml -DskipTests deploy
+  - BUILD_DOCS=1 VERSION_LIST="release-3.1 master" VERSION_ROOT="release-3.1"
```
Comment (joeybrk372) [inline .travis.yml]: I don't think you meant for `BUILD_DOCS` to be hardcoded

Comment (jharwig) [inline .travis.yml]: What do you mean?

Comment (jharwig) [inline .travis.yml]: That's what make the second job happen

Comment (joeybrk372) [inline .travis.yml]: it appears BUILD_DOCS is never changed, and you're else'ing the normal build

Comment (jharwig) [inline .travis.yml]: Each line in the env list is a new job in the build matrix. So the second has build docs set

Comment (jharwig) [inline .travis.yml]: The first line doesn't have it

Comment (joeybrk372) [inline .travis.yml]: ah I see, LGTM

**Pull request merged**